src/SFA.DAS.EmployerAccounts/Queries/GetClientContent/GetClientContentRequestHandler.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Threading.Tasks;
+using MediatR;
+using SFA.DAS.EmployerAccounts.Interfaces;
+using SFA.DAS.NLog.Logger;
+using SFA.DAS.Validation;
+
+namespace SFA.DAS.EmployerAccounts.Queries.GetClientContent
+{
+    public class GetClientContentRequestHandler : IAsyncRequestHandler<GetClientContentRequest, GetClientContentResponse>
+    {
+        private readonly IValidator<GetClientContentRequest> _validator;
+        private readonly ILog _logger;
+        private readonly IClientContentService _service;
+        private readonly ICacheStorageService _cacheStorageService;
+
+        public GetClientContentRequestHandler(
+            IValidator<GetClientContentRequest> validator,
+            ILog logger,
+            IClientContentService service, ICacheStorageService cacheStorageService)
+        {
+            _validator = validator;
+            _logger = logger;
+            _service = service;
+            _cacheStorageService = cacheStorageService;
+        }
+
+        public async Task<GetClientContentResponse> Handle(GetClientContentRequest message)
+        {
+            var validationResult = _validator.Validate(message);
+
+            if (!validationResult.IsValid())
+            {
+                throw new InvalidRequestException(validationResult.ValidationDictionary);
+            }
+
+            try
+            {
+                var type = (ContentType) Enum.Parse(typeof(ContentType), message.ContentType, true);
+
+                var cacheKey = message.ClientId;
+
+                if (!string.IsNullOrWhiteSpace(cacheKey))
+                {
+                    var result =
+                        await _cacheStorageService.RetrieveFromCache<string>(cacheKey);
+
+                    if (result != null)
+                    {
+                        return new GetClientContentResponse
+                        {
+
+                            ContentBanner = result
+                        };
+                    }
+                }
+                var cachedContentBanner = await _service.GetContentByClientId(type, message.ClientId);
+
+                if (cachedContentBanner == null)
+                {
+                    throw new CachedContentBannerNotFoundException(message.ClientId);
```

Comment:
Shouldn't the service throw a ContentNotFound exception or return a status that can be checked?

This error is misleading since the error doesn't happen because the cached item can't be found.